src/connector/SpectodaSimulatedConnector.ts

```
@@ -0,0 +1,654 @@
+  disconnect(): Promise<unknown> {
+    logging.verbose("disconnect()");
+
+    return new Promise(async (resolve, reject) => {
+      if (this.#connected) {
+        await sleep(100); // disconnecting logic process delay
+
+        this.#connected = false;
+        this.#runtimeReference.emit("#disconnected");
+      }
+      resolve(null); // always resolves even if there are internal errors
+    });
```

Biome (1.9.4)

[error] Promise executor functions should not be `async`.



(lint/suspicious/noAsyncPromiseExecutor)